operator/v1/types_csi_cluster_driver.go

```diff
@@ -35,40 +33,16 @@ type ClusterCSIDriver struct {
 	Status ClusterCSIDriverStatus `json:"status"`
 }
 
-// CSIDriverName is the name of the CSI driver
-// +kubebuilder:validation:Enum=ebs.csi.aws.com;manila.csi.openstack.org;csi.ovirt.org
-type CSIDriverName string
-
-// If you are adding a new driver name here, ensure that kubebuilder:validation:Enum is updated above
-// and 0000_90_cluster_csi_driver_01_config.crd.yaml-merge-patch file is also updated with new driver name.
-const (
-	AWSEBSCSIDriver CSIDriverName = "ebs.csi.aws.com"
-	ManilaCSIDriver CSIDriverName = "manila.csi.openstack.org"
-	OvirtCSIDriver  CSIDriverName = "csi.ovirt.org"
-)
-
```

Comment:
Couple of things as discussed in standup. 

1. We need these constants elsewhere and deleting them here will cause that code to break. Do we want to keep these constants here, so as each operator does not need to define individually?

2. Also - a casual reader of code may not understand, how different types can be disambiguated now. The fact that, they are disambiguated based on name sits outside this code and is not obvious. We need to document this at very minimum.

Reply:
cc @deads2k ^

Reply:
1. I restored the constants and `CSIDriverName` type.
2. I added "Name of the object must be name of the CSI driver it operates. See CSIDriverName type for list of allowed values." to `ClusterCSIDriver` description